pkg/cmd/cli/cmd/logs.go

```diff
@@ -189,5 +218,43 @@ func (o OpenShiftLogsOptions) RunLog() error {
 		// Use our own options object.
 		o.KubeLogOptions.Options = o.Options
 	}
-	return o.KubeLogOptions.RunLogs()
+	if !o.JenkinsLogs {
+		return o.KubeLogOptions.RunLogs()
+	}
+
+	bc, ok := o.KubeLogOptions.Object.(*buildapi.BuildConfig)
+	var urlString, buildName string
+	if !ok {
+		build, ok := o.KubeLogOptions.Object.(*buildapi.Build)
+		if ok {
+			buildName = build.ObjectMeta.Name
+			urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+			if len(urlString) == 0 {
+				return errors.New(fmt.Sprintf("The build %s did not have the correct log URL", buildName))
+			}
+		}
+	} else {
+		buildName = buildutil.BuildNameForConfigVersion(bc.ObjectMeta.Name, int(bc.Status.LastVersion))
+		build, _ := o.Client.Builds(o.Namespace).Get(buildName, metav1.GetOptions{})
+		if build == nil {
+			return errors.New(fmt.Sprintf("The build %s for build config %s was not found", buildName, bc.ObjectMeta.Name))
+		}
+		urlString, _ = build.Annotations[buildapi.BuildJenkinsBlueOceanLogURLAnnotation]
+		if len(urlString) == 0 {
+			return errors.New(fmt.Sprintf("The build %s did not have the correct log URL", buildName))
+		}
+	}
+	if !ok {
```

Comment:
In thinking how to adjust based on this comment,I realized I keep forgetting that golang's switch is much more powerful than java's and often cleaner vs. if-then-else chains.

Testing now ... will push what I think will be a more palatable implementation.